vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MergedRoomCreationItem.kt

```diff
@@ -107,13 +96,121 @@ abstract class MergedRoomCreationItem : BasedMergedItem<MergedRoomCreationItem.H
+                } else {
+                    holder.expandView.resources.getString(R.string.encryption_enabled_tile_description)
+                }
+                holder.e2eTitleDescriptionView.textAlignment = View.TEXT_ALIGNMENT_CENTER
+                holder.e2eTitleTextView.setCompoundDrawablesWithIntrinsicBounds(
+                        ContextCompat.getDrawable(holder.view.context, R.drawable.ic_shield_black),
+                        null, null, null
+                )
+            } else {
+                holder.e2eTitleTextView.text = holder.expandView.resources.getString(R.string.encryption_not_enabled)
+                holder.e2eTitleDescriptionView.text = holder.expandView.resources.getString(R.string.encryption_unknown_algorithm_tile_description)
+                holder.e2eTitleTextView.setCompoundDrawablesWithIntrinsicBounds(
+                        ContextCompat.getDrawable(holder.view.context, R.drawable.ic_shield_warning),
+                        null, null, null
+                )
+            }
+        } else {
+            holder.encryptionTile.isVisible = false
+        }
+    }
+
+    private fun bindCreationSummaryTile(holder: Holder) {
+        val roomSummary = attributes.roomSummary
+        val roomDisplayName = roomSummary?.displayName
+        holder.roomNameText.setTextOrHide(roomDisplayName)
+        val isDirect = roomSummary?.isDirect == true
+        val membersCount = roomSummary?.otherMemberIds?.size ?: 0
+
+        if (isDirect) {
+            holder.roomDescriptionText.text = holder.view.resources.getString(R.string.this_is_the_beginning_of_dm, roomSummary?.displayName ?: "")
+        } else if (roomDisplayName.isNullOrBlank() || roomSummary.name.isBlank()) {
+            holder.roomDescriptionText.text = holder.view.resources.getString(R.string.this_is_the_beginning_of_room_no_name)
+        } else {
+            holder.roomDescriptionText.text = holder.view.resources.getString(R.string.this_is_the_beginning_of_room, roomDisplayName)
+        }
+
+        val topic = roomSummary?.topic
+        if (topic.isNullOrBlank()) {
+            // do not show hint for DMs or group DMs
+            if (!isDirect) {
+                val addTopicLink = holder.view.resources.getString(R.string.add_a_topic_link_text)
+                val styledText = SpannableString(holder.view.resources.getString(R.string.room_created_summary_no_topic_creation_text, addTopicLink))
+                holder.roomTopicText.setTextOrHide(styledText.tappableMatchingText(addTopicLink, object : ClickableSpan() {
+                    override fun onClick(widget: View) {
+                        attributes.callback?.onTimelineItemAction(RoomDetailAction.QuickActionSetTopic)
+                    }
+                }))
+            }
+        } else {
+            holder.roomTopicText.setTextOrHide(span {
+                span(holder.view.resources.getString(R.string.topic_prefix)) {
+                    textStyle = "bold"
```

Comment:
Span should be built on controller if possible

Reply:
The thing is that the strings have links and are calling the callback, wouldn't it be strange to do that in the Controller?

Reply:
This is the goal of the controller, to prepare everything to be displayed. Items should be as simple as possible...